components/places/android/src/main/java/org/mozilla/places/PlacesConnection.kt

```diff
@@ -256,14 +332,43 @@ class SyncAuthInfo (
 )
 
 /**
- * An API for interacting with Places.
+ * An API for interacting with Places. This is the top-level entry-point, and
+ * exposes functions which return lower-level objects with the core
+ * functionality.
  */
-interface PlacesAPI {
+interface PlacesApiInterface {
     /**
-     * Record a visit to a URL, or update meta information about page URL. See [VisitObservation].
+     * Open a reader connection.
      */
-    fun noteObservation(data: VisitObservation)
+    fun openReader(): ReadablePlacesConnectionInterface
+
+    /**
+     * Open a writer connection.
+     *
+     * Note that this is not guaranteed to return a unique connection instance,
+     * and subsequent calls to getWriter may return the same connection
+     */
+    fun getWriter(): WritablePlacesConnectionInterface
+
+    /**
+     * Syncs the places stores.
+     *
+     * Note that this function blocks until the sync is complete, which may
+     * take some time due to the network etc. Because only 1 thread can be
```

Comment:
You mention threading in context of PlacesApi here. Can you expand on that, ideally in a docstring for PlacesApiInterface (or its impl..)? Quickly reading through this, I'm uncertain as to how this API should be used safely - e.g. threading distinction between an api object, specific writers, readers, etc.

Reply:
In general everything is thread safe, and locking is done internally.

I'll make it more clear when this happens, though.

Reply:
It's not clear to me how to rename these to follow `SpecificThing` vs `Thing` convention, so for now I guess the awkward names will stay.